src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -583,4 +588,82 @@ private void mapObservationUnitRows(
 			list.add(observationUnitRow);
 		}
 	}
+
+	@Override
+	public List<MeasurementVariable> getColumns(final int studyId, final int datasetId) {
```

Comment:
It seems this functionality is exactly the same as `org.ibp.api.java.impl.middleware.dataset.DatasetCSVExportServiceImpl#getColumns`
Perhaps we can extract it to a `datasetService.getAllVariables`
We should take any opportunity to refactor dataset services code as it's getting too complex, or we may end up with the same problem as in the old workbook services.

Reply:
It's the same. I will try to reuse it